libsolidity/ast/AsmJsonImporter.cpp

```diff
@@ -0,0 +1,276 @@
+}
+
+template <class T>
+T AsmJsonImporter::createAsmNode(Json::Value const& _node)
+{
+	// NOTE do I have to use boost::get? or can i just do it like that:
+	T r;
+	r.location = createSourceLocation(_node);
+	astAssert(!r.location.isEmpty() || !r.location.source, "Invalid source location in Asm AST");
+	return r;
+
+}
+
+Json::Value AsmJsonImporter::member(Json::Value const& _node, string const& _name)
+{
+	astAssert(_node.isMember(_name), "Node '" + _node["nodeType"].asString() +  " is missing field '" + _name + "'.");
+	return _node[_name];
+}
+
+TypedName AsmJsonImporter::createTypedName(Json::Value const& _node) {
+	TypedName typedName = createAsmNode<TypedName>(_node);
+	typedName.type = YulString{member(_node, "type").asString()};
+	typedName.name = YulString{member(_node, "name").asString()};
+	return typedName;
+}
+
+Statement AsmJsonImporter::createStatement(Json::Value const& _node)
+{
+	string nodeType = member(_node, "nodeType").asString();
+	astAssert(nodeType.substr(0, 3) == "Yul", "Invalid nodeType prefix");
+	nodeType = nodeType.substr(3);
+
+	if (nodeType == "ExpressionStatement")
+		return createExpressionStatement(_node);
+	else if (nodeType == "Assignment")
+		return createAssignment(_node);
+	else if (nodeType == "VariableDeclaration")
+		return createVariableDeclaration(_node);
+	else if (nodeType == "FunctionDefinition")
+		return createFunctionDefinition(_node);
+	else if (nodeType == "If")
+		return createIf(_node);
+	else if (nodeType == "Switch")
+		return createSwitch(_node);
+	else if (nodeType == "ForLoop")
+		return createForLoop(_node);
+	else if (nodeType == "Break")
+		return createBreak(_node);
+	else if (nodeType == "Continue")
+		return createContinue(_node);
+	else if (nodeType == "Leave")
+		return createLeave(_node);
+	else
+		astAssert(false, "Invalid nodeType as statement");
+}
+
+Expression AsmJsonImporter::createExpression(Json::Value const& _node)
+{
+	string nodeType = member(_node, "nodeType").asString();
+	astAssert(nodeType.substr(0, 3) == "Yul", "Invalid nodeType prefix");
+	nodeType = nodeType.substr(3);
+
+	if (nodeType == "FunctionCall")
+		return createFunctionCall(_node);
+	else if (nodeType == "Identifier")
+		return createIdentifier(_node);
+	else if (nodeType == "Literal")
+		return createLiteral(_node);
+	else
+		astAssert(false, "Invalid nodeType as expression");
+}
+
+vector<Expression> AsmJsonImporter::createExpressionVector(Json::Value const& _array)
+{
+	vector<Expression> ret;
+	for (auto& var: _array)
+		ret.emplace_back(createExpression(var));
+	return ret;
+}
+
+vector<Statement> AsmJsonImporter::createStatementVector(Json::Value const& _array)
+{
+	vector<Statement> ret;
+	for (auto& var: _array)
+		ret.emplace_back(createStatement(var));
+	return ret;
+}
+
+Block AsmJsonImporter::createBlock(Json::Value const& _node)
+{
+	Block block = createAsmNode<Block>(_node);
+	block.statements = createStatementVector(_node["statements"]); //NOTE use member helper here
```

Comment:
Is this note outstanding?

Reply:
I think the `create[..]` call is the helper